gnmpagerduty/tasks.py

```diff
@@ -0,0 +1,181 @@
+        super(HttpError,self).__init__(*args,**kwargs)
+        self.code = code
+        self.url = url
+        self.headers = headers
+        self.rtn_headers = rtn_headers
+        self.rtn_body = rtn_body
+
+    def __unicode__(self):
+        return u'HTTP error accessing {url}: {code}. Server said {rtn_body}'.format(
+            url=self.url,
+            code=self.code,
+            rtn_body=self.rtn_body
+        )
+
+    def __str__(self):
+        return self.__unicode__().encode('ascii')
+
+
+def make_vidispine_request(agent,method,urlpath,body,headers,content_type='application/xml'):
+    import base64
+    import re
+    from django.conf import settings
+    auth = base64.encodestring('%s:%s' % (settings.VIDISPINE_USERNAME, settings.VIDISPINE_PASSWORD)).replace('\n', '')
+
+    headers['Authorization']="Basic %s" % auth
+    headers['Content-Type']=content_type
+
+    if not re.match(r'^/',urlpath):
+        urlpath = '/' + urlpath
+
+    url = "{0}:{1}{2}".format(settings.VIDISPINE_URL,settings.VIDISPINE_PORT,urlpath)
+    logger.debug("URL is %s" % url)
+    (rtn_headers,content) = agent.request(url,method=method,body=body,headers=headers)
+    if int(rtn_headers['status']) < 200 or int(rtn_headers['status']) > 299:
+        raise HttpError(int(rtn_headers['status']),url,headers,rtn_headers,content)
+    return (rtn_headers,content)
+
+
+def get_system_type():
+    from django.core.cache import cache
+    import httplib2
+    system_type = cache.get('gnmpagerduty_system_type')
+    if system_type is None:
+        h=httplib2.Http()
+        headers, content = make_vidispine_request(h,'GET','/API/site?type=current',"",{'Accept': 'application/json'})
+
+        output = str(content)
+
+        if "KP" in output:
+            system_type = 'Live'
+        elif "BK" in output:
+            system_type = 'DC2'
+        elif "VX" in output:
+            system_type = 'Development'
+        else:
+            system_type = 'Unknown'
+        cache.set('gnmpagerduty_system_type', system_type)
+    return system_type
+
+
+@periodic_task(run_every=(crontab(minute='*/10')), name="check_storage", ignore_result=True)
+def check_storage():
+    from gnmvidispine.vs_storage import VSStoragePathMap
+    from django.conf import settings
+    from models import StorageData, IncidentKeys
+    import requests
+    import json
+
+    logger.info("check_storage run by Celery.")
+
+    i = app.control.inspect()
+
+    running = str(i.active())
+
+    rnumber = running.count("check_storage")
+
+    if rnumber > 1:
+        return "Task aborted to avoid duplication"
+
+    system_type = get_system_type()
+
+    storage_data = VSStoragePathMap(url=settings.VIDISPINE_URL,port=settings.VIDISPINE_PORT,
+                                    user=settings.VIDISPINE_USERNAME,passwd=settings.VIDISPINE_PASSWORD,
+                                    run_as=settings.VIDISPINE_USERNAME)
+    sd = {'data':'value', 'test':'value'}
```

Comment:
I mean, the 'data': 'value' and 'test':'value' contents.  My reading of the code suggested that we only ever access the 'map' attribute; in which case we can just assign the data that would be in sd['map'] to another variable name and not bother with the dictionary.